### PR TITLE
Adds overloads to GetArchive for adding custom timeouts 

### LINKED ISSUE #817: control request timeout when downloading archive
In #808 I set 1 hour as the request timeout when handling the redirect.

If you're going to want something different (for example, set it shorter to fail fast), you're going to want to specify overloads for `GetArchive`. I need to focus on more core concerns right now, so here's an opportunity for someone else to jump in and help out.

### PR DESCRIPTION
**Pull request by willsb**
This (hopefully) fixes #817

### CODE REVIEW

File: Octokit.Reactive/Clients/IObservableRepositoryContentsClient.cs [hunk 1]
```diff
@@ -97,6 +97,30 @@ public interface IObservableRepositoryContentsClient
...
+        /// Get an archive of a given repository's contents, in a specific format
+        /// </summary>
+        /// <remarks>https://developer.github.com/v3/repos/contents/#get-archive-link</remarks>
+        /// <param name="owner">The owner of the repository</param>
+        /// <param name="name">The name of the repository</param>
+        /// <param name="archiveFormat">The format of the archive. Can be either tarball or zipball</param>
+        /// <param name="reference">A valid Git reference.</param>
+        /// <param name="timeout"> Timeout in minutes </param>
+        /// <returns>The binary contents of the archive</returns>
+        IObservable<byte[]> GetArchive(string owner, string name, ArchiveFormat archiveFormat, string reference, int timeout);
```
Comment (Haacked) [inline Octokit.Reactive/Clients/IObservableRepositoryContentsClient.cs]: Honestly, I'd rather not have this method and only have the `TimeSpan` I think it promotes better practices. Also, if we really want this overload, we can add it as an extension method to the interface.

File: Octokit.Reactive/Clients/ObservableRepositoryContentsClient.cs [hunk 2]
```diff
@@ -166,6 +166,36 @@ public IObservable<RepositoryContent> GetAllContents(string owner, string name,
...
+        /// </summary>
+        /// <remarks>https://developer.github.com/v3/repos/contents/#get-archive-link</remarks>
+        /// <param name="owner">The owner of the repository</param>
+        /// <param name="name">The name of the repository</param>
+        /// <param name="archiveFormat">The format of the archive. Can be either tarball or zipball</param>
+        /// <param name="reference">A valid Git reference.</param>
+        /// <param name="timeout"> Time span until timeout </param>
+        /// <returns>The binary contents of the archive</returns>
+        public IObservable<byte[]> GetArchive(string owner, string name, ArchiveFormat archiveFormat, string reference, TimeSpan timeout)
+        {
```
Comment (Haacked) [inline Octokit.Reactive/Clients/ObservableRepositoryContentsClient.cs]: We should probably validate that the TimeSpan is not negative as a negative value is almost certainly a programmer error. I'd throw an `ArgumentOutOfRange` exception.

File: Octokit/Clients/RepositoryContentsClient.cs [hunk 3]
```diff
@@ -187,14 +187,44 @@ public Task<string> GetArchiveLink(string owner, string name, ArchiveFormat arch
...
+        /// <param name="owner">The owner of the repository</param>
+        /// <param name="name">The name of the repository</param>
+        /// <param name="archiveFormat">The format of the archive. Can be either tarball or zipball</param>
+        /// <param name="reference">A valid Git reference.</param>
+        /// <param name="timeout"> Time span until timeout </param>
+        /// <returns>The binary contents of the archive</returns>
+        public async Task<byte[]> GetArchive(string owner, string name, ArchiveFormat archiveFormat, string reference, TimeSpan timeout)
         {
             Ensure.ArgumentNotNullOrEmptyString(owner, "owner");
             Ensure.ArgumentNotNullOrEmptyString(name, "name");
```
Comment (Haacked) [inline Octokit/Clients/RepositoryContentsClient.cs]: Validate the `TimeSpan` argument here too please.

**Comment by willsb** [comment]
hmm, it seems like the build failed due to #904

File: Octokit.Reactive/Clients/ObservableRepositoryContentsClient.cs (hunk 2 above)
Comment (shiftkey) [inline Octokit.Reactive/Clients/ObservableRepositoryContentsClient.cs]: :thumbsup:

**Comment by shiftkey** [comment]
Apologies for the delay, and thanks for working on this!

**Pull request merged**